inc/ServerConfig.hpp

```diff
@@ -0,0 +1,25 @@
+#ifndef SERVERCONFIG_HPP
+#define SERVERCONFIG_HPP
+
+#include "Location.hpp"
+#include <iostream>
+#include <map>
+#include <vector>
+
+class ServerConfig {
+public:
+    ServerConfig();
+    ~ServerConfig();
+
+private:
+    std::string m_server_name;
+    std::string m_host;
+    std::string m_index;
+    std::string m_root;
```

Comment:
I think we should save every directive only in the lowest context. For root this would be "location". 

You can set root in server and in location context (also html). 
Since you don't need to set a location context at all, I think you would then save root in here. But this means that we have to check for every location if there is a root, and if not use the one from the server block. We also need to check if there is a location context and if not fall back.

In my mind:
- we set a standard location, if a server level root directive is declared. 
- In there we save the root directive, which is declared in server context
- If we have one (or more) location context defined, we set the root as declared in location context
- if not declared we set it as declared in server context
- if not declared we set it as default value. 

This way when we check which location should be used:
- if no location was defined we can still use our default location
- if one was declared, we check if it fits
- if root was also defined on server level, we create additional default location, which means we have two locations

I think the same stuff could also apply to other directives, but I don't see another one right now.